web/src/search_suggestion.js

```diff
@@ -750,7 +750,11 @@ export function get_search_result(base_query, query) {
     // is not displayed in that case. e.g. `messages with one or more abc` as
     // a suggestion for `has:abc`does not make sense.
     if (last.operator !== "" && last.operator !== "has" && last.operator !== "is") {
-        suggestion = get_default_suggestion(search_operators);
+        let last_suggestion_html = get_default_suggestion(last).description_html;
+        last_suggestion_html.replace(new RegExp('(' + last.operand + ')', 'ig'), function ($1, match) {
```

Comment:
Strings interpolated into a regexp need to be regexp-escaped.

```suggestion
        last_suggestion_html.replace(new RegExp('(' + _.escapeRegExp(last.operand) + ')', 'ig'), function ($1, match) {
```

Reply:
Our other highlighting for e.g. person names is done using `make_query_highlighter`, which looks safer.